### PR TITLE
Auth Implementation

### PR DESCRIPTION
**Pull request by horsemaker**
- Added Sign In Feature
- Added Sign Up Feature
- Refactored Home Screen & Header for newly implemented auth

Note: Sign Out Feature is not yet implemented.

recording

### CODE REVIEW

File: src/frontend/components/PrivateRoute/PrivateRoute.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { Navigate } from "react-router-dom";
+import { useAuth } from "../../contexts";
+
+const PrivateRoute = ({ children }) => {
+  const { auth } = useAuth();
+  return auth.status ? children : <Navigate replace to="/signin" />;
```
Comment (vijaytembugade) [inline src/frontend/components/PrivateRoute/PrivateRoute.js:7]: This is Nice, Got to know now!

File: src/frontend/constants/auth-constants.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export const KAFKA_AUTH_USER_TOKEN = "KAFKA_AUTH_USER_TOKEN";
```
Comment (vijaytembugade) [inline src/frontend/constants/auth-constants.js:1]: You can name it AUTH_USER_TOKEN only, but its also okay

File: src/frontend/layouts/Main/Main.js [hunk 3]
```diff
@@ -1,15 +1,40 @@
...
+          <>
+            <Route path="/signin" element={<SignInScreen />} />
+            <Route path="/signup" element={<SignUpScreen />} />
+          </>
+        )}
+        <Route
+          path="/notes"
+          element={
+            <PrivateRoute>
+              <></>
```
Comment (vijaytembugade) [inline src/frontend/layouts/Main/Main.js:25]: empty?

File: src/frontend/services/auth-services/signUpService.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import axios from "axios";
+
+export const signUpService = async (user) => {
+  try {
+    const { data } = await axios.post("/api/auth/signup", user);
+    console.log(data);
```
Comment (vijaytembugade) [inline src/frontend/services/auth-services/signUpService.js]: you can remove this console.log

**Review by vijaytembugade — APPROVED** [review]
Good work Yash. Loved the theme and style. 
image

File: src/frontend/layouts/Main/Main.js (hunk 3 above)
Comment (horsemaker) [reply src/frontend/layouts/Main/Main.js:25]: Have not made the screen yet.

File: src/frontend/services/auth-services/signUpService.js (hunk 4 above)
Comment (horsemaker) [reply src/frontend/services/auth-services/signUpService.js]: Right!

File: src/frontend/contexts/auth-context.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { useState, createContext, useContext } from "react";
+import { KAFKA_AUTH_USER_DATA, KAFKA_AUTH_USER_TOKEN } from "../constants";
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const [auth, setAuth] = useState({
+    status: localStorage.getItem(KAFKA_AUTH_USER_TOKEN) ? true : false,
+    token: localStorage.getItem(KAFKA_AUTH_USER_TOKEN),
+    user: JSON.parse(localStorage.getItem(KAFKA_AUTH_USER_DATA)),
```
Comment (t007rushi) [inline src/frontend/contexts/auth-context.js:10]: can put in variable just suggestion

File: src/frontend/screens/SignUpScreen/SignUpScreen.js [hunk 6]
```diff
@@ -58,8 +112,20 @@ export const SignUpScreen = () => {
...
+                  ...user,
+                  confirmPassword: e.target.value,
+                })
+              }
             />
           </div>
+          {password !== "" &&
+            confirmPassword !== "" &&
+            password !== confirmPassword && (
+              <small className="error-message">Passwords don't match!</small>
```
Comment (t007rushi) [inline src/frontend/screens/SignUpScreen/SignUpScreen.js:127]: nice check

**Review by t007rushi — APPROVED** [review]
Overall UI is 🔥

**Pull request merged**